common/components/ProductDetail.jsx

```diff
@@ -5,7 +5,7 @@ import * as ShopActions from '../actions/ShopActions';
 import { Link } from 'react-router';
 import { fetchNeeds } from '../utils/fetchComponentData';
 
-export default class ProductDetail extends Component {
+export class ProductDetail extends Component {
```

Comment:
You can actually remove the export as well. Only the redux connected version needs to be exposed.